src/state_machine.zig

```diff
@@ -337,151 +331,161 @@ pub const StateMachine = struct {
-            if (!equal_32_bytes(exists.reserved, t.reserved)) {
-                return .exists_with_different_reserved_field;
+
+            var dr = self.get_account(lookup.debit_account_id) orelse return .debit_account_not_found;
+            var cr = self.get_account(lookup.credit_account_id) orelse return .credit_account_not_found;
+            assert(lookup.timestamp > dr.timestamp);
+            assert(lookup.timestamp > cr.timestamp);
+
+            assert(lookup.flags.pending);
+            if (dr.debits_pending < lookup.amount) return .debit_amount_was_not_reserved;
+            if (cr.credits_pending < lookup.amount) return .credit_amount_was_not_reserved;
+
+            // Once reserved, the amount can be moved from reserved to accepted without breaking limits:
+            assert(!dr.debits_exceed_credits(0));
+            assert(!cr.credits_exceed_debits(0));
+
+            // TODO We can combine this lookup with the previous lookup if we return `error!void`:
+            var insert = self.posted.getOrPutAssumeCapacity(t.id);
+            if (insert.found_existing) {
+                unreachable;
+            } else {
+                insert.value_ptr.* = t;
+                dr.debits_pending -= lookup.amount;
+                cr.credits_pending -= lookup.amount;
+                if (!t.flags.void_pending_transfer) {
+                    if (t.amount == 0) {
+                        dr.debits_posted += lookup.amount;
+                        cr.credits_posted += lookup.amount;
+                    } else {
+                        dr.debits_posted += t.amount;
+                        cr.credits_posted += t.amount;
+                    }
+                }
+                self.post_timestamp = t.timestamp;
+                return .ok;
             }
-            if (exists.timeout != t.timeout) return .exists_with_different_timeout;
-            return .exists;
         } else {
+            if (t.flags.padding != 0) return .reserved_flag_padding;
+            if (t.flags.pending) {
+                // Otherwise reserved amounts may never be released:
+                if (t.timeout == 0) return .two_phase_commit_must_timeout;
+            } else if (t.timeout != 0) {
+                return .timeout_reserved_for_two_phase_commit;
+            }
+            if (!t.flags.hashlock and !zeroed_32_bytes(t.reserved)) return .reserved_field;
+
+            if (t.amount == 0) return .amount_is_zero;
+
+            if (t.debit_account_id == t.credit_account_id) return .accounts_are_the_same;
+
+            // The etymology of the DR and CR abbreviations for debit and credit is interesting, either:
+            // 1. derived from the Latin past participles of debitum and creditum, debere and credere,
+            // 2. standing for debit record and credit record, or
+            // 3. relating to debtor and creditor.
+            // We use them to distinguish between `cr` (credit account), and `c` (commit).
```

Comment:
No, `dr` and `cr` are domain-specific shorthand for debit and credit account respectively (it's literally how you indicate the accounts in a double-entry journal entry), so they should stay.

We can drop this part of the comment: `// We use them to distinguish between `cr` (credit account), and `c` (commit).` since that's no longer relevant.